### ISSUE TITLE
PYI033 False positive

### ISSUE THREAD

**Issue by twoertwein** [charliermarsh/ruff#4151]
test.pyi :
`# pyright: reportIncompleteStub = false`

```sh
$ ruff --version
ruff 0.0.263

$ ruff check --isolated --select PYI033 test.pyi
test.pyi:1:1: PYI033 Don't use type comments in stub file
Found 1 error.
```

xref https://github.com/pandas-dev/pandas/pull/52974

**Comment by charliermarsh** [comment]
Thanks!

**Comment by evanrittenhouse** [comment]
Looks like our typing regex flags if the beginning is `type` or `pyright` (https://github.com/charliermarsh/ruff/blob/main/crates/ruff/src/rules/flake8_pyi/rules/type_comment_in_stub.rs#L53) but doesn't check if there's truly a Python type after it.

**Comment by twoertwein** [comment]
> Looks like our typing regex flags if the beginning is `type` or `pyright`

flake8-pyi doesn't seem to search for `"pyright"`: https://github.com/PyCQA/flake8-pyi/search?q=pyright

**Comment by evanrittenhouse** [comment]
@charliermarsh I can raise a PR for this if we want to remove that check, but I'm not sure why we included it in the first place and don't want to unintentionally break something. 

Alternatively, pyright sets flags via `=` - we could alter the regex to only emit if `type` is present or `pyright` is present and `=` is not. Please let me know which you'd prefer!

**Comment by charliermarsh** [comment]
I think it was a mistake to include `pyright` in this specific rule, since `# pyright: ` comments are only used for ignores and a few other directives, and not as "type comments".

**Comment by charliermarsh** [comment]
So we should just remove it.

**Comment by evanrittenhouse** [comment]
Perfect, that's what I was leaning towards as well.
